src/misc/lv_anim.c

```
@@ -499,6 +500,30 @@ uint32_t lv_anim_resolve_speed(uint32_t speed_or_time, int32_t start, int32_t en
     return LV_CLAMP(min_time * 10, time, max_time * 10);
 }
 
+bool lv_anim_is_paused(lv_anim_t * a)
+{
+    return a->is_paused;
+}
+
+void lv_anim_pause(lv_anim_t * a)
```

What happens if `a` is NULL? Is guaranteed this will never occur? Additionally since this a function that does not return value, I would like to see at least a LOG_ERR/WRN when things get wrong, it would facillitate and avoid silent failures.

---

If we really want  to make sure that `a` is valid, we would need to go through the animation list to find `a`

---

There are some asserting macros for NULL you can use on these cases, my intention here is just to provide some hint for the user  if something goes wrong, making easier to debug when the test coverage does not reach some potential edge case.

---

Yep, I agree with you